lib/akami/wsse.rb

```diff
@@ -91,25 +91,24 @@ def body_attributes
 
     # Returns the XML for a WSSE header.
     def to_xml
-      if signature? and signature.have_document?
-        Gyoku.xml wsse_signature.merge!(hash)
-      elsif username_token? && timestamp?
-        Gyoku.xml wsse_username_token.merge!(wsu_timestamp) {
-          |key, v1, v2| v1.merge!(v2) {
-            |key, v1, v2| v1.merge!(v2)
-          }
-        }
-      elsif username_token?
-        Gyoku.xml wsse_username_token.merge!(hash)
-      elsif timestamp?
-        Gyoku.xml wsu_timestamp.merge!(hash)
-      else
-        ""
-      end
+      h = wsse_signature if signature? && signature.have_document?
+      h = merge_hashes_with_keys(h, wsse_username_token) if username_token?
+      h = merge_hashes_with_keys(h, wsu_timestamp) if timestamp?
+
+      return '' unless h
+      Gyoku.xml h
     end
 
   private
 
+    def merge_hashes_with_keys(hash_one, hash_two)
```

Comment:
There's ` if username_token?` and `if timestamp?` on the called, seems like an overlap in responsibility?